app/src/features/Preview/Preview.tsx

```diff
@@ -93,6 +107,13 @@ const Preview = (): JSX.Element => {
     sourceId?: string;
     mappingId?: string;
   }>();
+  const { data: mapping } = useApiResourcesRetrieveQuery({
+    id: mappingId ?? "",
+  });
```

Comment:
You can skip this request if mappingId is undefined

```suggestion
  },
  {
    skip: !Boolean(mappingId),
  }
  );
```